service/src/main/kotlin/fi/espoo/evaka/reports/ServiceVoucherValueReport.kt

```diff
@@ -100,24 +113,35 @@ data class ServiceVoucherValueRow(
     val childFirstName: String,
     val childLastName: String,
     val childDateOfBirth: LocalDate,
+    val childGroupName: String?,
     val unitId: UUID,
     val unitName: String,
     val areaId: UUID,
     val areaName: String,
     val serviceVoucherPeriod: Period,
     val serviceVoucherValue: Int,
     val serviceVoucherCoPayment: Int,
-    val serviceVoucherServiceCoefficient: Int
+    val derivatives: ValueRowDerivatives? = null
+)
+
+data class ValueRowDerivatives(
+    val realizedAmount: Int,
+    val coefficient: BigDecimal,
+    val realizedPeriod: ClosedPeriod,
+    val numberOfDays: Long
```

Comment:
Also, this is a matter of taste, but these could all just be "computed" values on `ServiceVoucherValueRow`, like this:

```
data class ServiceVoucherValueRow(
  ...,
  monthPeriod: ClosedPeriod
) {
  @JsonProperty("realizedPeriod")
  fun realizedPeriod(): ClosedPeriod = ClosedPeriod(maxOf(serviceVoucherPeriod.start, monthPeriod.start), minOf(serviceVoucherPeriod.end ?: monthPeriod.end, monthPeriod.end))

  @JsonProperty("numberOfDays")
  fun numberOfDays() = realizedPeriod.durationInDays().toInt()

  ...
}
```

Reply:
Hmm yep they would apart from `realizedPeriod` which needs `monthPeriod` from surrounding context :/

Reply:
I'm pretty late with this comment, but that's why I included `monthPeriod` in the class 😄

Anyway, @reynders needs to approve this so it can be merged.

Reply:
Aaa yes now I see that `monthPeriod` :D Oh dear. Will fix this.